cv-data-service-library/src/main/java/com/trihydro/library/helpers/CdotGisConnector.java

```
@@ -0,0 +1,74 @@
+
+  @Autowired
+  public void InjectDependencies(RestTemplateProvider _restTemplateProvider) {
+    this.restTemplateProvider = _restTemplateProvider;
+  }
+
+  public String getBaseUrl() {
+    return baseUrl;
+  }
+
+  public RestTemplateProvider getRestTemplateProvider() {
+    return restTemplateProvider;
+  }
+
+  public ResponseEntity<String> getAllRoutes() {
+    String targetUrl = baseUrl + "/Routes";
+    logger.info("Getting all routes from CDOT GIS service at: {}", targetUrl);
+    String params = "?f=json";
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Accept", "application/json");
+    HttpEntity<String> entity = new HttpEntity<>(headers);
+    ResponseEntity<String> response;
+    try {
+      response = restTemplateProvider.GetRestTemplate().exchange(targetUrl + params, HttpMethod.GET, entity, String.class);
+    } catch (Exception e) {
+        logger.error("Error getting all routes from CDOT GIS service: {}", e.getMessage(), e);
+        return null;
+    }
+    if (response.getBody() != null) {
+      logger.info("Received response from CDOT GIS service: {}", response.getBody());
+    }
+    return response;
+  }
+
+  public ResponseEntity<String> getRouteById(String routeId) {
+    String targetUrl = baseUrl + "/Route";
+    logger.info("Getting route with ID {} from CDOT GIS service at: {}", routeId, targetUrl);
+    String params = "?routeId=" + routeId + "&outSR=4326&f=json";
+    HttpHeaders headers = new HttpHeaders();
+    headers.set("Accept", "application/json");
+    HttpEntity<String> entity = new HttpEntity<>(headers);
+    ResponseEntity<String> response;
+    try {
+      response = restTemplateProvider.GetRestTemplate().exchange(targetUrl + params, HttpMethod.GET, entity, String.class);
+    } catch (Exception e) {
+        logger.error("Error getting route with ID {} from CDOT GIS service: {}", routeId, e.getMessage(), e);
+        return null;
+    }
```

question(blocking): why can't we throw the exception up to the caller, and let them handle it? Returning null opens us up to NullPointerExceptions, and is generally not a safe practice https://medium.com/javarevisited/just-dont-return-null-dcdf5d77128f

---

I think throwing the exception is a great idea! Transitioned to throwing exceptions instead of returning null in CdotGisConnector and CdotUpstreamPathController.